Source/WTF/wtf/text/WTFString.h

```diff
@@ -512,6 +514,18 @@ inline void String::clearImplIfNotShared()
         m_impl = nullptr;
 }
 
+template<typename Func>
+inline Expected<std::invoke_result_t<Func, Span<const char>>, UTF8ConversionError> String::tryGetUtf8ForRange(const Func& function, unsigned offset, unsigned length, ConversionMode mode) const
+{
+    ASSERT(offset <= this->length());
+    ASSERT(offset + length <= this->length());
```

Comment:
This was of writing the assertion is vulnerable to overflow if `length` is very large. Not critical but better to write it in a way that is not. One way is `ASSERT(length <= this->length() - offset)` but I am sure there are many others.

Reply:
Fixed.